src/control/server/security_rpc.go

```diff
@@ -64,7 +72,7 @@ func (m *SecurityModule) processValidateCredentials(body []byte) ([]byte, error)
 	}
 
 	// Check our verifier
-	err = auth.VerifyToken(key, cred.GetToken(), cred.GetVerifier().GetData())
+	err = auth.VerifyToken(key, cred.GetToken(), cred.GetVerifier().GetData(), m.validAuthFlavors)
```

Comment:
Could remove the parameter from VerifyToken and do the check of "is this flavor valid" in the body of this function.